clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackSecurityGroup.groovy

```diff
@@ -37,8 +42,64 @@ class OpenstackSecurityGroup implements SecurityGroup {
   final Set<Rule> inboundRules
   final Set<Rule> outboundRules
 
+  @JsonIgnore
   @Override
   SecurityGroupSummary getSummary() {
     new OpenstackSecurityGroupSummary(name: name, id: id)
   }
+
+  static OpenstackSecurityGroup from(SecGroupExtension securityGroup, String accountName, String region) {
+    new OpenstackSecurityGroup(id: securityGroup.id,
+      accountName: accountName,
+      region: region,
+      name: securityGroup.name,
+      description: securityGroup.description,
+      inboundRules: buildInboundRules(securityGroup, accountName, region)
+    )
+  }
+
+  private static List<Rule> buildInboundRules(SecGroupExtension securityGroup, String accountName, String region) {
+    securityGroup.rules.collect { rule ->
+      // The Openstack4J library doesn't put a type on the rule, instead, it includes a range object with a null cidr
+      rule.range?.cidr ? buildIpRangeRule(rule) : buildSecurityGroupRule(rule, accountName, region)
+    }
+  }
+
+  private static SecurityGroupRule buildSecurityGroupRule(SecGroupExtension.Rule rule, String accountName, String region) {
+    def portRange = new Rule.PortRange(startPort: rule.fromPort, endPort: rule.toPort)
+    def securityGroup = new OpenstackSecurityGroup(
+      name: rule.group.name,
+      type: OpenstackCloudProvider.ID,
+      accountName: accountName,
+      region: region
+    )
+    new SecurityGroupRule(protocol: rule.IPProtocol.value(),
+      portRanges: [portRange] as SortedSet,
+      securityGroup: securityGroup
+    )
+  }
+
+  private static IpRangeRule buildIpRangeRule(SecGroupExtension.Rule rule) {
+    def portRange = new Rule.PortRange(startPort: rule.fromPort, endPort: rule.toPort)
+    def addressableRange = buildAddressableRangeFromCidr(rule.range.cidr)
+    new IpRangeRule(protocol: rule.IPProtocol.value(),
+      portRanges: [portRange] as SortedSet,
+      range: addressableRange
+    )
+  }
+
+  private static AddressableRange buildAddressableRangeFromCidr(String cidr) {
+    if (!cidr) {
+      return null
+    }
+
+    def rangeParts = cidr.split('/') as List
+
+    // If the cidr just a single IP address, use 32 as the mask
+    if (rangeParts.size() == 1) {
```

Comment:
I had it elsewhere and moved it here as a static method. I thought I needed it in 2 locations, but I guess I only needed it in one place, but kept it as a static method anyways. I had pulled this logic from the Google security group caching.